v3-sdk/minting-position/src/example/Example.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import './Example.css'
+import { ethers } from 'ethers'
+import {
+  Pool,
+  computePoolAddress,
+  Position,
+  nearestUsableTick,
+  NonfungiblePositionManager,
+} from '@uniswap/v3-sdk'
+import { Percent } from '@uniswap/sdk-core'
+import { Environment, CurrentConfig } from '../config'
+import {
+  getCurrencyBalance,
+  getPositionIds,
+  getTokenTransferApprovals,
+} from '../libs/contracts'
+import {
+  connectBrowserExtensionWallet,
+  getProvider,
+  TransactionState,
+  sendTransaction,
+  getWalletAddress,
+} from '../libs/providers'
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json'
+import {
+  POOL_FACTORY_CONTRACT_ADDRESS,
+  MAX_FEE_PER_GAS,
+  MAX_PRIORITY_FEE_PER_GAS,
+  NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS,
+} from '../libs/constants'
+
+const getPoolContract = async (): Promise<ethers.Contract> => {
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error('No provider')
+  }
+
+  const currentPoolAddress = computePoolAddress({
+    factoryAddress: POOL_FACTORY_CONTRACT_ADDRESS,
+    tokenA: CurrentConfig.tokens.in,
+    tokenB: CurrentConfig.tokens.out,
+    fee: CurrentConfig.tokens.fee,
+  })
+
+  return new ethers.Contract(
+    currentPoolAddress,
+    IUniswapV3PoolABI.abi,
+    provider
+  )
+}
+
+const getPoolConstants = async (): Promise<{
+  token0: string
+  token1: string
+  fee: number
+  tickSpacing: number
+}> => {
+  const poolContract = await getPoolContract()
+
+  const [token0, token1, fee, tickSpacing] = await Promise.all([
+    poolContract.token0(),
+    poolContract.token1(),
+    poolContract.fee(),
+    poolContract.tickSpacing(),
+  ])
+
+  return {
+    token0,
+    token1,
+    fee,
+    tickSpacing,
+  }
+}
+
+const getCurrentPoolState = async (): Promise<{
+  sqrtPriceX96: ethers.BigNumber
+  liquidity: ethers.BigNumber
+  tick: number
+}> => {
+  const poolContract = await getPoolContract()
+
+  const [liquidity, slot] = await Promise.all([
+    poolContract.liquidity(),
+    poolContract.slot0(),
+  ])
+
+  return {
+    liquidity,
+    sqrtPriceX96: slot[0],
+    tick: slot[1],
+  }
+}
```

Comment:
I've combined these on mine into a single `getPoolInfo` function. I can see the argument for separating, but I'm curious how you feel about that combination. It's a lot shorter codewise (saves a duplicated `getPoolContract` call + multiple `Promise.all`'s) and we can explain that some properties are constant while other is state in guides / comments

Reply:
Sounds good